From d8d87fab0861e478e3f03ee43c1731124d00784f Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 5 Oct 2015 23:20:54 -0400
Subject: [PATCH 10/10] Set a more reasonable netty recycler capacity

Default Netty Recycler capacity is 262k~, resulting in major memory
consumption over long runtimes that will never free.

This lowers it by 80%, which should only even be hit on the largest of servers.

Copied from https://github.com/WaterfallMC/Waterfall/commit/c6783ef7e958195035cedbfc182b9838da6c8db3

Signed-off-by: Janmm14 <computerjanimaus@yahoo.de>
---
 bootstrap/src/main/java/net/md_5/bungee/BungeeCordLauncher.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/bootstrap/src/main/java/net/md_5/bungee/BungeeCordLauncher.java b/bootstrap/src/main/java/net/md_5/bungee/BungeeCordLauncher.java
index f130ac4..6626ec5 100644
--- a/bootstrap/src/main/java/net/md_5/bungee/BungeeCordLauncher.java
+++ b/bootstrap/src/main/java/net/md_5/bungee/BungeeCordLauncher.java
@@ -20,6 +20,12 @@ public class BungeeCordLauncher
         Security.setProperty( "networkaddress.cache.ttl", "30" );
         Security.setProperty( "networkaddress.cache.negative.ttl", "10" );
 
+        // TODO: remove .default for netty 5!
+        if ( System.getProperty( "io.netty.recycler.maxCapacity.default" ) == null )
+        {
+            System.setProperty( "io.netty.recycler.maxCapacity.default", "50000" );
+        }
+
         OptionParser parser = new OptionParser();
         parser.allowsUnrecognizedOptions();
         parser.acceptsAll( Arrays.asList( "v", "version" ) );
-- 
1.9.5.msysgit.0

